elasticdl/pkg/common/tensor.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"bytes"
+	"elasticdl.org/elasticdl/pkg/proto"
+	"encoding/binary"
+	"math"
+)
+
+// Tensor defines tensor struct
+// TODO(qijun): handle different tensor dtype
+type Tensor struct {
+	Name    string
+	Value   []float32
+	Dim     []int64
+	Indices []int64
+}
+
+// GetTensorSize returns the size of a tensor
+func GetTensorSize(t *Tensor) int64 {
```

Comment:
We can use method of struct instead:
```
func (t *Tensor)Size() int64 {
   ...
}
```
Then for a `tensor`, call `tensor.Size()` to get its size.

Reply:
Rename It to `GetDimProduct`, so it could be used for any slice.